firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts

```diff
@@ -100,22 +100,71 @@ export class FirestoreBigQueryEventHistoryTracker
     return data;
   }
 
+  /**
+   * Check whether a failed operation is retryable or not.
+   * Reasons for retrying:
+   * 1) We added a new column to our schema. Sometimes BQ is not ready to stream insertion records immediately
+   *    with the new column.
+   */
+  private async isRetryableInsertionError(e) {
+    let isRetryable = true;
+    const expectedErrors = [
+      {
+        "message": "no such field.",
+        "location": "document_id"
+      }
+    ]
+    if (
+      e.response &&
+      e.response.insertErrors &&
+      e.response.insertErrors.errors
+    ) {
+      const errors = e.response.insertErrors.errors;
+      errors.map((error) => {
+        let isExpected = false;
+        expectedErrors.map(expectedError => {
+          if (error.message === expectedError.message && error.location === expectedError.location) {
+            isExpected = true;
+          }
+        })
+        if (!isExpected) {
+          isRetryable = false;
+        }
+      });
+    }
+    return isRetryable;
+  }
+
   /**
    * Inserts rows of data into the BigQuery raw change log table.
    */
-  private async insertData(rows: bigquery.RowMetadata[]) {
+  private async insertData(
+    rows: bigquery.RowMetadata[],
+    overrideOptions: any,
+    retry: boolean
+  ) {
     const options = {
       skipInvalidRows: false,
       ignoreUnknownValues: false,
       raw: true,
     };
+    for (const key in overrideOptions) {
+      options[key] = overrideOptions[key];
+    }
```

Comment:
nit: these lines can be consolidated with the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals):

```ts
const options = {
  skipInvalidRows: false,
  ignoreUnknownValues: false,
  raw: true,
  ...overrideOptions,
};
```